lib/mongoid/delorean/trackable.rb

```diff
@@ -59,83 +58,77 @@ def without_history_tracking
       end
 
       def revert!(version = (self.version - 1))
-        old_version = self.versions.where(version: version).first
+        old_version = versions.where(version: version).first
         if old_version
           old_version.full_attributes.each do |key, value|
-            self.write_attribute(key, value)
+            write_attribute(key, value)
           end
-          self.save!
+          save!
         end
       end
 
       module CommonEmbeddedMethods
-
         def save_version
-          if self._parent.respond_to?(:save_version)
-            if self._parent.respond_to?(:track_history?)
-              if self._parent.track_history?
-                self._parent.save_version
-              end
+          if _parent.respond_to?(:save_version)
+            if _parent.respond_to?(:track_history?)
+              _parent.save_version if _parent.track_history?
             else
-              self._parent.save_version
+              _parent.save_version
             end
           end
 
           true
         end
-
       end
 
       module CommonInstanceMethods
-
         def changes_with_relations
-          _changes = self.changes.dup
+          _changes = changes.dup
 
-          %w{version updated_at created_at}.each do |col|
+          %w[version updated_at created_at].each do |col|
             _changes.delete(col)
             _changes.delete(col.to_sym)
           end
 
           relation_changes = {}
-          self.embedded_relations.each do |name, details|
-            relation = self.send(name)
+          embedded_relations.each do |name, details|
+            relation = send(name)
             relation_changes[name] = []
-            if details.relation == Mongoid::Relations::Embedded::One
+            if details.relation == Mongoid::Association::Embedded::EmbedsOne::Proxy
```

Comment:
Check if `details.is_a? Mongoid::Association::Embedded::EmbedsOne` instead of checking for Proxy class

Reply:
changed